internal/journal/volumegroupjournal.go

```diff
@@ -115,14 +115,14 @@ func (sgj *volumeGroupJournalConfig) Connect(
 	monitors,
 	namespace string,
 	cr *util.Credentials,
-) (*volumeGroupJournalConfig, error) {
```

Comment:
we might get into problem when we serve many requests for different pools because all the journaling interface are initialized once and when are using different pool or ceph users we get connection for each one and that's the reason for returning which helps in initializing the connections, we cannot call Connect for same global object from multiple places. for now lets keep this and adjust it later or remove connect from the interface and modify it to take the input struct and return interface from it.

Reply:
Currently it returns a pointer to itself, not a new instance of `volumeGroupJournalConfig`.

Why would someone not create multiple VolumeGroupJournal instances for different pools? If the connection/credentials to the Ceph cluster are the same, they will share the ClusterConnection already.

Reply:
probably we might need to address it when required, For now approving it as there is no consumer of this yet